### PR TITLE
Whole File Reads, Fixes #17

### LINKED ISSUE #17: Add a "full text" codec
We should add a codec that grabs the full text of a file and (maybe) a filter that can remove XML tags.

This would allow the plugin to index things like text files, source code, or static sites hosted on Cloud Storage.

### PR DESCRIPTION
**Pull request by josephlewis42**
Allow the plugin to read whole files from the bucket, this can lead to some future demos like anonymizing the text, machine language detection/translation, image tagging or STT.

Perhaps a good path to a `logstash-filter-google_ai`.

### CODE REVIEW

File: beater/codec/blob_test.go [hunk 1]
```diff
@@ -0,0 +1,96 @@
...
+import (
+	"strings"
+
+	"testing"
+)
+
+func TestBlobCodecNextErr(t *testing.T) {
+	cases := map[string]struct {
+		Data      string
+		Length    int
```
Comment (rosbo) [inline beater/codec/blob_test.go]: Consider renaming to `ExpectLength` or `ExpectedLength` to make it clear that this field is an expected value and is not used as an input for the test.

File: beater/codec/blob_test.go [hunk 2]
```diff
@@ -0,0 +1,96 @@
...
+
+	if !codec.Next() {
+		t.Error("Quit too early.")
+	}
+
+	val := codec.Value()
+
+	// lines are indexed at 1
+	if val["line"] != 1 {
+		t.Errorf("Expected line to be %d, got %d", val["line"], 1)
```
Comment (rosbo) [inline beater/codec/blob_test.go]: I think the parameters are reversed.

File: beater/codec/blob_test.go [hunk 3]
```diff
@@ -0,0 +1,96 @@
...
+	}
+
+	if val["event"] != expected {
+		t.Errorf("Expected event to be %q, got %q", expected, val["event"])
+	}
+
+	if codec.Next() {
+		t.Error("Should only be able to call next once")
+	}
+
```
Comment (rosbo) [inline beater/codec/blob_test.go]: Rm extra line

File: beater/codec/clob_test.go [hunk 4]
```diff
@@ -0,0 +1,95 @@
...
+
+	if !codec.Next() {
+		t.Error("Quit too early.")
+	}
+
+	val := codec.Value()
+
+	// lines are indexed at 1
+	if val["line"] != 1 {
+		t.Errorf("Expected line to be %d, got %d", val["line"], 1)
```
Comment (rosbo) [inline beater/codec/clob_test.go]: Parameters reversed

**Review by rosbo — CHANGES_REQUESTED** [review]

**Comment by josephlewis42** [comment]
@rosbo thanks for the review! I think they are all fixed now.

**Review by rosbo — APPROVED** [review]

**Pull request merged**